### PR TITLE
spring-boot-prac-shoppingCart

### PR DESCRIPTION
**Pull request by Nonzeq**

### CODE REVIEW

File: src/main/java/com/kobylchak/bookstore/mapper/CartItemMapper.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    @AfterMapping
+    default void setBook(
+            @MappingTarget CartItem cartItem,
+            @Context BookRepository bookRepository,
+            CartItemRequestDto requestDto) {
+        Book book = bookRepository.findById(requestDto.getBookId())
+                            .orElseThrow(() -> new EntityNotFoundException("Book not found"));
+        cartItem.setBook(book);
+
```
Comment (MAMentorFX) [inline src/main/java/com/kobylchak/bookstore/mapper/CartItemMapper.java]:
```suggestion
```

File: src/main/java/com/kobylchak/bookstore/service/cart/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        ShoppingCart shoppingCart = cartItemService.updateItem(requestDto,
+                                                               getShoppingCartByUser(),
+                                                               id);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCart getShoppingCartByUser() {
+        String email = SecurityContextHolder.getContext().getAuthentication().getName();
```
Comment (MAMentorFX) [inline src/main/java/com/kobylchak/bookstore/service/cart/ShoppingCartServiceImpl.java]: You shouldn't use Infrastructure code here, you shoud pass Authentication to each controller's method, and pass user (retrieved from aunthetication) to service

File: src/main/java/com/kobylchak/bookstore/service/cart/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartDto updateItemInShoppingCart(CartItemRequestUpdateDto requestDto, Long id) {
+        ShoppingCart shoppingCart = cartItemService.updateItem(requestDto,
+                                                               getShoppingCartByUser(),
+                                                               id);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
```
Comment (MAMentorFX) [inline src/main/java/com/kobylchak/bookstore/service/cart/ShoppingCartServiceImpl.java]: Why are those Transactional, you put trasactional if class has more than one call to DB

File: src/main/java/com/kobylchak/bookstore/service/cart/item/CartItemServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    public ShoppingCart addItem(CartItemRequestDto requestDto, ShoppingCart shoppingCart) {
+        CartItem cartItem = cartItemMapper.toModel(requestDto, bookRepository, shoppingCart);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItemRepository.save(cartItem);
+        shoppingCart.getCartItems().add(cartItem);
+        return shoppingCart;
+    }
+
+    @Override
+    public ShoppingCart updateItem(
```
Comment (MAMentorFX) [inline src/main/java/com/kobylchak/bookstore/service/cart/item/CartItemServiceImpl.java:35]:
```suggestion
@Transactional
    public ShoppingCart updateItem(
```

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**